### PR TITLE
Support write to local file instead of post on github

### PR DESCRIPTION
**Pull request by NikitaCOEUR**
#194 

## Why
I have also the requirement to obtain the results of the Terraform plan/apply parsed and nicely formatted by tfcmt. So I am trying (for the first time) to contribute to this enhancement with my proposal.
The old method is still fully functional, I have just added a new parameter "output" to specify a file to write the result to.

## To use it : 
Simply use it same way before but now we can do this : 
### plan
```
tfcmt --output plan.md plan -- terraform plan -no-color
```
image

### apply 
```
tfcmt --output apply.md apply-- terraform apply-no-color
```
image

------------
I attempted to preserve the essence of what I saw in the code.
I hope it will be useful and accurate... Don't hesitate to provide feedback as this is my first time working with Go.

### CODE REVIEW

**Comment by suzuki-shunsuke** [comment]
Thank you for your contribution!

**Review by suzuki-shunsuke — COMMENTED** [review]
The package name `localFile` isn't good.

https://go.dev/blog/package-names

>  They are lower case, with no under_scores or mixedCaps.

File: pkg/config/config.go [hunk 1]
```diff
@@ -23,6 +23,7 @@ type Config struct {
 	PlanPatch          bool   `yaml:"plan_patch"`
 	RepoOwner          string `yaml:"repo_owner"`
 	RepoName           string `yaml:"repo_name"`
+	Output			   string `yaml:"output_file"`
```
Comment (suzuki-shunsuke) [inline pkg/config/config.go:26]:
```suggestion
	Output			   string `yaml:"-"`
```

Hmm. I think `output_file` should be specified by command line argument explicitly.

File: pkg/config/config.go [hunk 2]
```diff
@@ -107,6 +108,10 @@ func (cfg *Config) LoadFile(path string) error {
 
 // Validate validates config file
 func (cfg *Config) Validate() error {
+
```
Comment (suzuki-shunsuke) [inline pkg/config/config.go:111]:
```suggestion

```

**Review by suzuki-shunsuke — COMMENTED** [review]
It isn't bad to implement Notifier for local file, but I wonder if there is better approach. 🤔
Maintaining both github and local file isn't desirable.

**Comment by NikitaCOEUR** [comment]
> It isn't bad to implement Notifier for local file, but I wonder if there is better approach. 🤔 Maintaining both github and local file isn't desirable.

I also wondered about it, but I wanted to reuse as much existing code as possible without having to refactor everything. In the "controller.go" file, we return "client.Notify", so I wanted to keep that intact. That's why I decided to create a new notifier. Do you have a better suggestion?

**Comment by suzuki-shunsuke** [comment]
> > It isn't bad to implement Notifier for local file, but I wonder if there is better approach. 🤔 Maintaining both github and local file isn't desirable.
> 
> I also wondered about it, but I wanted to reuse as much existing code as possible without having to refactor everything. In the "controller.go" file, we return "client.Notify", so I wanted to keep that intact. That's why I decided to create a new notifier. Do you have a better suggestion?

I see. Thank you for your explanation. As you said, we shouldn't change existing code as much as possible in this pull request.
We can refactor after merging this pull request.

**Comment by suzuki-shunsuke** [comment]
Please fix CI. https://github.com/suzuki-shunsuke/tfcmt/actions/runs/4174408268/jobs/7228404508

**Comment by NikitaCOEUR** [comment]
Using underscores in package names is not allowed, so I renamed "local_file" to "tofile". I find "tofile" to be a more preferable name than "localfile". What is your opinion on this?

**Comment by suzuki-shunsuke** [comment]
I prefer `localfile` than `tofile` because `localfile` is more consistent with the Notifier package `github`.

**Comment by suzuki-shunsuke** [comment]
I'm not sure why GitHub Actions wasn't triggered. 🤔 
I couldn't approve the workflow run in UI.

image

**Comment by NikitaCOEUR** [comment]
Uh... Maybe my force push ? I've amended a commit yesterday.

**Comment by NikitaCOEUR** [comment]
Have you try it via actions tab ?

File: pkg/notifier/localfile/output.go [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+		return err
+	}
+
+	defer file.Close()
+
+	_, err2 := file.WriteString(body)
+
+	if err2 != nil {
+		return err2
+	}
```
Comment (suzuki-shunsuke) [inline pkg/notifier/localfile/output.go:29]:
```suggestion
	if _, err := file.WriteString(body); err != nil {
		return err
	}
```

**Comment by suzuki-shunsuke** [comment]
I added a commit, then I can approve and run.

image

**Review by suzuki-shunsuke — APPROVED** [review]
Thanks!

**Comment by suzuki-shunsuke** [comment]
https://github.com/suzuki-shunsuke/tfcmt/issues/194#issuecomment-1435661832

**Pull request merged**